src/backends/mod.rs

```diff
@@ -119,25 +119,20 @@ pub trait Backend: Clone {
     /// It recursively does the computation down the stream until it is not a symbolic variable.
     /// - The symbolic variables are stored in the witness_vars.
     /// - The computed values are stored in the cached_values.
-    fn compute_var(&self, env: &mut WitnessEnv<Self::Field>, var: CellVar) -> Result<Self::Field> {
-        self.compute_val(env, self.witness_vars(var), var.index)
-    }
-
-    fn compute_val(
+    fn compute_var(
         &self,
-        env: &mut WitnessEnv<Self::Field>,
-        val: &Value<Self>,
-        var_index: usize,
+        env: &mut WitnessEnv<Self::Field, Self::CellVar>,
+        var: Self::CellVar,
     ) -> Result<Self::Field> {
-        if let Some(res) = env.cached_values.get(&var_index) {
+        if let Some(res) = env.cached_values.get(&var) {
```

Comment:
I'm a bit confused as how this cache would work now, mmmm

Reply:
reverted to using usize: [2721919](https://github.com/zksecurity/noname/pull/69/commits/2721919849348ace25988f6346b470b69ce7d86d)